services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -1227,6 +1229,22 @@ Optional<String> getTopicForCurrentPushJob(
           }
           return Optional.empty();
         } else {
+          Map<String, ControllerClient> controllerClients = getVeniceHelixAdmin().getControllerClientMap(clusterName);
+          for (Map.Entry<String, ControllerClient> entry: controllerClients.entrySet()) {
+            ControllerClient controllerClient = entry.getValue();
+            StoreResponse response = controllerClient.getStore(storeName);
+            if (!response.isError()) {
+              StoreInfo store = response.getStore();
+              Optional<Version> version = store.getVersion(versionNumber);
+              // if there is a online future version available, do not allow further pushes
+              if (version.isPresent() && version.get().getStatus().equals(ONLINE)
+                  && version.get().isVersionSwapDeferred() && store.getCurrentVersion() < versionNumber) {
+                throw new VeniceException(
```

Comment:
BTW, the previous branch return the topic name, shall we keep a consistent behavior here?